app/src/main/java/org/mozilla/focus/tiles/CustomTilesAccessor.kt

```diff
@@ -31,15 +31,50 @@ data class HomeTile (
             jsonObject.getString(KEY_ID))
 }
 
-object HomeTilesManager {
+/**
+ * Static accessor of custom home tiles, that is backed by SharedPreferences.
+ *
+ * New sites are appended to the end of the list.
+ *
+ * This keeps a cached version of the custom home tiles that have been pinned,
+ * in order to be more performant when checking whether sites are pinned or not.
+ * In order to keep the cache consistent, should only be called from the UIThread.
+ */
+object CustomTilesAccessor {
+    private lateinit var customTilesCache: LinkedHashMap<String, JSONObject>
```

Comment:
nit: it feels weird that this cache isn't `LinkedHashMap<String, HomeTile>`. Trade-offs:
- As HomeTile, it's type-safe and the data-type we want to manipulate ("more correct")
- Converting to JSONObjects can be expensive, but we don't do that much anyway (and we can cache those values in HomeTile if we wanted)
- Currently, we're allocating a bunch of HomeTile objects each time the home screen is shown, which can be expensive if it forces GC